sway-core/src/asm_lang/mod.rs

```diff
@@ -1279,19 +1305,22 @@ pub(crate) enum ControlFlowOp<Reg> {
     // Just a comment that will be inserted into the asm without an op
     Comment,
     // Jumps to a label
-    Jump(Label),
-    // Jumps to a label if the register is not equal to zero
-    JumpIfNotZero(Reg, Label),
-    // Jumps to a label, similarly to Jump, though semantically expecting to return.
-    Call(Label),
+    Jump {
+        /// Target label
+        to: Label,
+        /// Jump type
+        type_: JumpType<Reg>,
+        /// Force far jump, reserving space for an extra opcode to load target address.
+        /// If this is false but the target is far, [`mark_far_jumps`] pass will enable this.
+        /// Also, this will be enabled for self-jumps, as they require a noop to be inserted before them.
+        force_far: bool,
```

Comment:
Introducing a global (i.e., accessible everywhere the instruction can be) flag seems kind of ugly. Can you not just keep a `FxHashSet` of jumps requiring `force_far`? This can be created in `realize_labels`, and you pass it to `resolve_labels` as a mutable parameter  (or alternatively `resolve_labels` computes and returns it) and then directly used within the function when calling `compile_jump`.

Reply:
Done in 75df97093